code/__HELPERS/unsorted.dm

```diff
@@ -1594,6 +1594,15 @@ atom/proc/GetTypeInAllContents(typepath)
 			colour += temp_col
 	return colour
 
+/proc/get_random_chemical(var/is_plant = 0)
+	var/list/blocked = blocked_chems		//blocked_chems list is found in code/_globalvars/lists/reagents.dm

```

Comment:
This is just creating a reference to blocked_chems- don't think that's what you intend. If you want a copy of the list, change this to `var/list/blocked = blocked_chems.Copy()`